motoman_driver/MotoPlus/IoServer.c

```diff
@@ -324,26 +328,30 @@ int Ros_IoServer_ReadIOBit(SimpleMsg* receiveMsg, SimpleMsg* replyMsg)
 	replyMsg->header.msgType = ROS_MSG_MOTO_READ_IO_BIT_REPLY;
 	replyMsg->header.commType = ROS_COMM_SERVICE_REPLY;
 	
-	ioReadInfo.ulAddr = receiveMsg->body.readIOBit.ioAddress;
-	apiRet = mpReadIO(&ioReadInfo, &ioValue, 1);
+	if (Ros_IoServer_IsValidReadAddress(receiveMsg->body.readIOBit.ioAddress, IO_ACCESS_BIT))
+	{
+		ioReadInfo.ulAddr = receiveMsg->body.readIOBit.ioAddress;
+		apiRet = mpReadIO(&ioReadInfo, &ioValue, 1);
```

Comment:
Is `1` the number of addressed to read? If so: could we avoid the magic nr and use a named constant?

Reply:
But magic is fun...

Reply:
Certainly :)

Reply:
This has been updated